### ISSUE TITLE
Allow forecast.lm to handle fourier terms

### ISSUE THREAD

**Issue by robjhyndman** [robjhyndman/forecast#60]
This should work
```
gas <- ts(read.csv("http://robjhyndman.com/data/gasoline.csv", header=FALSE)[,1], 
          freq=365.25/7, start=1991+31/7/365.25)
fit <- tslm(gas ~ trend + fourier(gas, 3))
forecast(fit, newdata=fourierf(gas, 3, 52))
```

**Comment by robjhyndman** [comment]
Unfortunately, predict.lm is very temperamental. This is the best I can do for now:
```
forecast(fit, newdata=data.frame(X=I(fourierf(gas,3,52))))
```

**Comment by mitchelloharawild** [comment]
From what I've found so far, this is less of a scoping issue, but more of a naming issue, so predict.lm cannot find the variables it needs. Instead, it looks at the formula variables and evaluates them in the parent environment, giving the result of fourier(gas, 3) as newdata, not fourierf(gas, 3, 52)... Hence the row mismatch warning:
```
Warning messages:
1: 'newdata' had 52 rows but variables found have 745 rows 
2: 'newdata' had 52 rows but variables found have 745 rows 
```
The forecast.lm error appears to be a quick-fix error message for when predict.lm doesn't correctly find the model variables.
The mismatch for the above example can be seen here:
```
colnames(newdata)
[1] "S1.52"  "C1.52"  "S2.52"  "C2.52"  "S3.52"  "C3.52"  "trend"  "season"

colnames(object$model)
[1] "gas"             "fourier(gas, 3)"
```
object$model$`fourier(gas, 3)` does contain the variables named as above ({S,C}1:3.52), however they are not found by predict.lm. There are a couple of ways I can think to solve this...

The "quick-fix" would be to evaluate and flatten any matrix objects before they are passed to lm() and change the formula. So instead of calling 
```R
lm(gas ~ trend + fourier(gas, 3))
```
We would call (by changing the formula during tslm).
```R
lm(gas ~ trend + S1.52 + C1.52 + S2.52 + C2.52 + S3.52 + C3.52))
```
Each time we call predict.lm, we would need to ensure that the data has been flattened in forecast.lm.

The more difficult, but probably better solution (structurally), is to maintain the data.frame matrix structure for the functions when calling lm, and have the resulting newdata format to have consistent colnames as the model's variables. So we would detect fourierf(gas, 3, 52) as intending to replace data for fourier(gas, 3) and rename it accordingly, so predict.lm can find its data. I believe this would work in a more general sense, and accordingly have less bugs associated down the line.

Which approach do you think is best?

**Comment by mitchelloharawild** [comment]
Fixed in #207

**Comment by mitchelloharawild** [comment]
@robjhyndman 
Resolved in #207.
